src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,325 @@
+        const systems = await this.querySystems(filterStr, defaultProjection);
+        utilization_array_with_alias = utilization_array.map(system_utilization_df => {
+          const foundItem = systems.find((system: SystemMetadata) => system.id === system_utilization_df.id);
+          return { ...system_utilization_df, alias: foundItem ? foundItem.alias : 'System' };
+        });
+      } else {
+        const filterArr: string[] = [];
+        idsArray.forEach((id) => {
+          filterArr.push(`${AssetFilterProperties.AssetIdentifier} == "${id}"`);
+        });
+        const filterStr = workspace_id ? ` (${filterArr.join(' or ')}) and workspace = "${workspace_id}"` : filterArr.join(' or ');
+        const assetsResponse = await this.queryAssets(filterStr, -1);
+        utilization_array_with_alias = utilization_array.map(asset_utilization_df => {
+          const foundItem = assetsResponse.find((asset: AssetModel) => asset.id === asset_utilization_df.id);
+          return { ...asset_utilization_df, alias: foundItem ? foundItem.name : 'Asset' };
+        });
+      }
+      const suffix = `${isPeakLabels[query.isPeak]} ${timeFrequencyLabels[query.timeFrequency]}`;
+      result.fields = [
+        { name: 'time', values: utilization_array_with_alias[0].datetimes },
+      ];
+      utilization_array_with_alias.forEach((data) => {
+        result.fields.push(
+          {
+            name: data.id,
+            values: data.values,
+            // default configuration for visualizations
+            config: {
+              displayName: `${data.alias} ${suffix}`,
+              unit: '%',
+              min: 0,
+              max: 100
+            }
+          }
+        )
+      })
+      return result;
+    }
+  }
+
+  shouldRunQuery(_: AssetQuery): boolean {
+    return true;
+  }
+
+  private async fetchAndProcessUtilizationData(query: AssetQuery, options: DataQueryRequest): Promise<Array<{
+    id: string,
+    datetimes: number[],
+    values: number[]
+  }>> {
+    const [from, to] = [options.range.from.valueOf(), options.range.to.valueOf()];
+    const workspace_id = this.templateSrv.replace(query.workspace);
+    const minion_id = this.templateSrv.replace(query.minionId);
+    // this value is set in the Datasource -> Query options -> Max data points
+    const maxDataPoints = options.maxDataPoints;
+    let continuationToken = '';
+    const {
+      assetIdentifier,
+      minionId,
+      utilizationCategory,
+      isPeak,
+      peakDays,
+      timeFrequency
+    } = query;
+    const workingHoursPolicy = {
+      'startTime': momentToTime(dateTime(query.peakStart)),
+      'endTime': momentToTime(dateTime(query.nonPeakStart))
+    }
+    const utilizationFilter: string = utilizationCategory === UtilizationCategory.TEST ? `Category = \"Test\" or Category = \"test\"` : '';
+    let arrayOfEntityIds: string[];
```

Comment:
```suggestion
    let entityIds: string[];
```

We don't usually put the type of a variable in its name